internal/server/analytics/sink_test.go

```diff
@@ -0,0 +1,80 @@
+package analytics
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/attribute"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+	"go.uber.org/zap"
+)
+
+type sampleSink struct {
+	ch chan []*EvaluationResponse
+}
+
+func (s *sampleSink) IncrementFlagEvaluationCounts(ctx context.Context, responses []*EvaluationResponse) error {
+	go func() {
+		s.ch <- responses
+	}()
```

Comment:
Curiosity: Does this need to be done in a goroutine, or does the otel trace provider already do this asynchronously from your code? My gut would've guessed that this is already performed in another goroutine and therefor won't block if you just send this inline. However, I could be mistaken.

If I am, you could also get away without doing this by giving the sample sink channel a non-zero buffer. That is a take it or leave it though.

Reply:
@GeorgeMac Yeah so OTEL actually calls `ExportSpans` synchronously: https://pkg.go.dev/go.opentelemetry.io/otel/sdk/trace#SpanExporter

Which in turn would call this function. So when `span.End()` is called here: https://github.com/flipt-io/flipt/blob/337b179050f7fe2f14a75839aa79b5e49d4ed016/internal/server/analytics/sink_test.go#L63

If there is no channel waiting to receive at the time it is invoked it will block.